src/main/webapp/app/components/tabs/CurationToolsTab.tsx

```diff
@@ -0,0 +1,75 @@
+import { PAGE_ROUTE } from 'app/config/constants/constants';
+import { Gene, Meta } from 'app/shared/model/firebase/firebase.model';
+import GeneSelect from 'app/shared/select/GeneSelect';
+import { getFirebasePath } from 'app/shared/util/firebase/firebase-utils';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+import React, { useEffect, useState } from 'react';
+import { FaExclamationCircle } from 'react-icons/fa';
+import { Button, Col, Row } from 'reactstrap';
+import './curation-tools-tab.scss';
+
+function CurationToolsTab({ addMetaListener, metaData, createMetaGene, createGene }: StoreProps) {
+  const [selectedGene, setSelectedGene] = useState<string>(null);
+  const [createButtonDisabled, setCreateButtonDisabled] = useState(true);
+
+  useEffect(() => {
+    const callback = addMetaListener();
+
+    return () => callback && callback();
+  }, []);
+
+  useEffect(() => {
+    if (selectedGene && !Object.keys(metaData).includes(selectedGene)) {
+      setCreateButtonDisabled(false);
+    } else {
+      setCreateButtonDisabled(true);
+    }
+  }, [selectedGene]);
+
+  async function handleCreateGene() {
+    const gene = new Gene(selectedGene);
+    await Promise.all([
+      createGene(getFirebasePath('GENE', gene.name), gene),
+      createMetaGene(getFirebasePath('META_GENE', gene.name), new Meta()),
+    ]);
+    window.location.href = `${PAGE_ROUTE.CURATION}/${gene.name}`;
+  }
+
+  return (
+    <div>
+      <Row className="mb-2">
+        <Col className="px-0">
+          <h6>Curate New Gene</h6>
+        </Col>
+      </Row>
+      <Row className="mb-2">
+        <Col className="px-0">
+          <GeneSelect onChange={option => setSelectedGene(option?.label)} />
+        </Col>
+      </Row>
+      <Row className={`align-items-center ${createButtonDisabled && selectedGene ? 'justify-content-between' : 'justify-content-end'}`}>
+        {createButtonDisabled && selectedGene && (
```

Comment:
The logic of when`createButtonDisabled && selectedGene`, then gene already exists should be a computed value. Or probably just rename `createButtonDisabled`?

Reply:
I changed the component quite a bit. `selectedGene` is now a ref since it changing does not affect the UI (the GeneSelect handles its own rendering). 

`createButtonDisabled` and `showGeneExistsWarning` now are the only stateful variables. They both need to exist since if a gene is not selected, the button should be disabled, but the warning should not be shown.